From 492f998f6ff3e07e3dd7148780bf267cb3214f8d Mon Sep 17 00:00:00 2001
From: Benjamin Beaudry <benjamin.beaudry@microsoft.com>
Date: Tue, 07 May 2024 12:16:13 +0000
Subject: [PATCH] [a11y] Replace CHECK by DWC in AXObject::UpdateChildrenIfNecessary

This CL downgrades a CHECK to a DUMP_WILL_BE_CHECK to reduce the
impact on our users. We know this is hit but we're not certain why, so
it will give us the time to fix it while preventing crashes.

Bug: 331297887
Change-Id: I6909736b23f2b1712d067587e7a8e1ec7cf7a72e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5519626
Auto-Submit: Benjamin Beaudry <benjamin.beaudry@microsoft.com>
Reviewed-by: Aaron Leventhal <aleventhal@chromium.org>
Commit-Queue: Aaron Leventhal <aleventhal@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1297422}
---

--- a/third_party/blink/renderer/modules/accessibility/ax_object.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_object.cc
@@ -6054,10 +6054,13 @@
     return;
   }
 
-  CHECK(!AXObjectCache().IsFrozen())
-      << "Object should have already had its children updated in "
-         "AXObjectCacheImpl::UpdateTreeIfNeeded(): "
-      << ToString(true, true);
+  if (AXObjectCache().IsFrozen()) {
+    DUMP_WILL_BE_CHECK(!AXObjectCache().IsFrozen())
+        << "Object should have already had its children updated in "
+           "AXObjectCacheImpl::UpdateTreeIfNeeded(): "
+        << ToString(true, true);
+    return;
+  }
 
   if (!CanHaveChildren()) {
     // Clear any children in case the node previously allowed children.
